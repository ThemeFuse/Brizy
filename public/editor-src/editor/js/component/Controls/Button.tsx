import React, { ReactElement } from "react";
import classNames from "classnames";
import { WithClassName } from "visual/utils/options/attributes";
import EditorIcon from "visual/component/EditorIcon";

export interface Props extends WithClassName {
  onClick: VoidFunction;
  reverse?: boolean;
  label?: string;
  icon?: string;
  align?: "left" | "center" | "right";
}

export const Button = ({
  className,
  onClick,
  icon,
  label,
  reverse,
  align = "center"
}: Props): ReactElement | null => {
  return label || icon ? (
    <div
      className={classNames(
        "brz-ed-control__button",
        `brz-ed-control__button-${align}`,
        className,
        {
          reverse: !!reverse
        }
      )}
      onClick={onClick}
    >
      {icon && <EditorIcon icon={icon} />}
      <span className={"brz-ed-control__label"}>{label}</span>
    </div>
  ) : null;
};
