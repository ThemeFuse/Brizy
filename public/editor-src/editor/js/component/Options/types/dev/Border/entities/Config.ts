import { Style } from "./style";
import { WidthType } from "./widthType";

export interface Config {
  opacity?: boolean;
  styles?: Style[];
  width?: WidthType[];
}
