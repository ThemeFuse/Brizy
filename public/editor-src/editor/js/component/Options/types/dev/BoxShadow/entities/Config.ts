import { Type } from "./Type";

export interface Config {
  type?: Exclude<Type, "none">;
  opacity?: boolean;
}
