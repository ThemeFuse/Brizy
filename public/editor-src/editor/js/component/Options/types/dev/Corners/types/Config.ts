import { Unit } from "./Unit";

export interface Config {
  units?: Unit[];
}
