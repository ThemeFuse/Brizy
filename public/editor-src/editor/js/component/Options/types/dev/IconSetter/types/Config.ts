export interface Config {
  canDelete?: boolean;
}
