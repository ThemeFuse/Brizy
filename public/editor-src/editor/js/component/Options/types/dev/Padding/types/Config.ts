import { SpacingUnit } from "visual/component/Options/utils/SpacingUnit";

export interface Config {
  units?: SpacingUnit[];
}
