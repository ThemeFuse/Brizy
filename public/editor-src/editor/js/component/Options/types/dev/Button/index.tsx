import React from "react";
import * as Option from "visual/component/Options/Type";
import { Button as Control } from "visual/component/Controls/Button";
import { WithConfig } from "visual/utils/options/attributes";

export interface Config {
  text: string;
  icon: string;
  reverse?: boolean;
}

export interface Props extends Option.Props<undefined>, WithConfig<Config> {
  onClick: VoidFunction;
}

export const Button: React.FC<Props> & Option.OptionType<undefined> = ({
  label,
  onClick,
  config
}) => {
  return (
    <>
      {label}
      <Control
        onClick={onClick}
        label={config?.text}
        icon={config?.icon}
        reverse={config?.reverse}
      />
    </>
  );
};

const getModel: Option.FromElementModel<undefined> = () => undefined;
const getElementModel: Option.ToElementModel<undefined> = () => ({});

Button.fromElementModel = getModel;
Button.toElementModel = getElementModel;

// @ts-expect-error: Variable 'defaultValue' implicitly has an 'any' type
Button.defaultValue = undefined;
