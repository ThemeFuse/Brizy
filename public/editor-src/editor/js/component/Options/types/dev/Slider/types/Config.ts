import { Unit } from "visual/component/Controls/NumberUnit/types";
import { Size } from "visual/types/attributes";

export type Config = {
  min?: number;
  max?: number;
  step?: number;
  debounceUpdate?: boolean;
  updateRate?: number;
  inputMin?: number;
  inputMax?: number;
  units?: Unit<string>[];
  size?: Size;
};
