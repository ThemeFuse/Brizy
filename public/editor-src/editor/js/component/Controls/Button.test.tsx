import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Button } from "./Button";
import Config from "visual/global/Config";

describe("Test Button", () => {
  beforeAll(() => {
    global.IS_EDITOR = true;

    Config.load({
      // @ts-expect-error to find out what happens here
      urls: {
        editorIcons: "/"
      }
    });
  });

  const onClick = jest.fn();

  describe("Snapshots", () => {
    test("If not label or icon provided, do not render anything", () => {
      const button = render(<Button onClick={onClick} />);

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("With icon", () => {
      const button = render(<Button onClick={onClick} icon={"nc-cog"} />);

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("With label", () => {
      const button = render(<Button onClick={onClick} label={"Test"} />);

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("With label add icon", () => {
      const button = render(
        <Button onClick={onClick} label={"Test"} icon={"nc-cog"} />
      );

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("With label and icon and align Left", () => {
      const button = render(
        <Button
          onClick={onClick}
          label={"Test"}
          icon={"nc-cog"}
          align={"left"}
        />
      );

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("With label and icon and align Right", () => {
      const button = render(
        <Button
          onClick={onClick}
          label={"Test"}
          icon={"nc-cog"}
          align={"left"}
        />
      );

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("Reversed", () => {
      const button = render(
        <Button
          onClick={onClick}
          label={"Test"}
          icon={"nc-cog"}
          reverse={true}
        />
      );

      expect(button.container.firstChild).toMatchSnapshot();
    });
  });

  test("Test onClick", () => {
    const button = render(
      <Button onClick={onClick} icon={"nc-cog"} label={"Test"} />
    );
    const input = button.container.firstElementChild;

    input && fireEvent.click(input);

    expect(onClick).toBeCalledTimes(1);
  });
});
