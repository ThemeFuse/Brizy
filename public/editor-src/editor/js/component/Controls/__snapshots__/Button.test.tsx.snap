// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Snapshots If not label or icon provided, do not render anything 1`] = `null`;

exports[`Test Button Snapshots Reversed 1`] = `
<div
  class="brz-ed-control__button brz-ed-control__button-center reverse"
>
  <svg
    class="brz-icon-svg brz-ed-icon-svg"
  >
    <use
      xlink:href="//icons.svg#nc-cog"
    />
  </svg>
  <span
    class="brz-ed-control__label"
  >
    Test
  </span>
</div>
`;

exports[`Test Button Snapshots With icon 1`] = `
<div
  class="brz-ed-control__button brz-ed-control__button-center"
>
  <svg
    class="brz-icon-svg brz-ed-icon-svg"
  >
    <use
      xlink:href="//icons.svg#nc-cog"
    />
  </svg>
  <span
    class="brz-ed-control__label"
  />
</div>
`;

exports[`Test Button Snapshots With label 1`] = `
<div
  class="brz-ed-control__button brz-ed-control__button-center"
>
  <span
    class="brz-ed-control__label"
  >
    Test
  </span>
</div>
`;

exports[`Test Button Snapshots With label add icon 1`] = `
<div
  class="brz-ed-control__button brz-ed-control__button-center"
>
  <svg
    class="brz-icon-svg brz-ed-icon-svg"
  >
    <use
      xlink:href="//icons.svg#nc-cog"
    />
  </svg>
  <span
    class="brz-ed-control__label"
  >
    Test
  </span>
</div>
`;

exports[`Test Button Snapshots With label and icon and align Left 1`] = `
<div
  class="brz-ed-control__button brz-ed-control__button-left"
>
  <svg
    class="brz-icon-svg brz-ed-icon-svg"
  >
    <use
      xlink:href="//icons.svg#nc-cog"
    />
  </svg>
  <span
    class="brz-ed-control__label"
  >
    Test
  </span>
</div>
`;

exports[`Test Button Snapshots With label and icon and align Right 1`] = `
<div
  class="brz-ed-control__button brz-ed-control__button-left"
>
  <svg
    class="brz-icon-svg brz-ed-icon-svg"
  >
    <use
      xlink:href="//icons.svg#nc-cog"
    />
  </svg>
  <span
    class="brz-ed-control__label"
  >
    Test
  </span>
</div>
`;
